pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -1169,3 +1170,249 @@ func awsProviderSpecFromMachine(mapiMachine *machinev1beta1.Machine) (machinev1b
+
+			// Add multiple forbidden fields in one update
+			awsMachine.Spec.ImageLookupFormat = testImageLookupFormat
+			awsMachine.Spec.NetworkInterfaces = []string{testNetworkInterface}
+			awsMachine.Spec.UncompressedUserData = ptr.To(true)
+
+			err := k8sClient.Update(ctx, awsMachine)
+			Expect(err).To(HaveOccurred())
+			// Should catch the first forbidden field (validation stops at first error)
+			expectVAPError(err, "spec.imageLookupFormat is a forbidden field")
+		})
+
+		It("should not enforce the VAP on other namespaces", func() {
+			otherNamespace := corev1resourcebuilder.Namespace().WithGenerateName("other-namespace").Build()
+			Expect(k8sClient.Create(ctx, otherNamespace)).To(Succeed())
+
+			awsMachine := awsv1resourcebuilder.AWSMachine().WithGenerateName("test-aws-machine").WithNamespace(otherNamespace.Name).Build()
+			awsMachine.Spec.ImageLookupFormat = testImageLookupFormat
+			err := k8sClient.Create(ctx, awsMachine)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+	})
+
+	Context("AWSMachineTemplate validation", func() {
+		const (
+			testImageLookupFormat = "ami-format"
+			testImageLookupOrg    = "123456789012"
+			testImageLookupBaseOS = "linux"
+			testSecretPrefix      = "my-secret"
+			testSecurityGroupID   = "sg-123"
+			testNetworkInterface  = "eni-12345678"
+			testVaultBackend      = "vault"
+		)
+
+		DescribeTable("should validate AWSMachineTemplate creation",
+			func(name string, modifier func(*awsv1.AWSMachineTemplate), expectedError string) {
+				awsMachineTemplate := awsv1resourcebuilder.AWSMachineTemplate().WithGenerateName("test-aws-machine-template").WithNamespace(namespace.Name).Build()
+
+				if modifier != nil {
+					modifier(awsMachineTemplate)
+				}
+
+				err := k8sClient.Create(ctx, awsMachineTemplate)
+
+				if expectedError != "" {
+					Expect(err).To(HaveOccurred())
+					expectVAPError(err, expectedError)
+				} else {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			},
+			Entry("without forbidden fields", "", nil, ""),
+			Entry("with a forbidden field (ami.eksLookupType)", "ami.eksLookupType", func(mt *awsv1.AWSMachineTemplate) {
+				mt.Spec.Template.Spec.AMI.EKSOptimizedLookupType = ptr.To(awsv1.AmazonLinux)
+			}, "spec.ami.eksLookupType is a forbidden field"),
+			Entry("with a forbidden field (imageLookupFormat)", "imageLookupFormat", func(mt *awsv1.AWSMachineTemplate) {
+				mt.Spec.Template.Spec.ImageLookupFormat = testImageLookupFormat
+			}, "spec.imageLookupFormat is a forbidden field"),
+			Entry("with a forbidden field (imageLookupOrg)", "imageLookupOrg", func(mt *awsv1.AWSMachineTemplate) {
+				mt.Spec.Template.Spec.ImageLookupOrg = testImageLookupOrg
+			}, "spec.imageLookupOrg is a forbidden field"),
+			Entry("with a forbidden field (imageLookupBaseOS)", "imageLookupBaseOS", func(mt *awsv1.AWSMachineTemplate) {
+				mt.Spec.Template.Spec.ImageLookupBaseOS = testImageLookupBaseOS
+			}, "spec.imageLookupBaseOS is a forbidden field"),
```

Comment:
Can we use messageExpression and a variable to make sure these errors are actually pathing to the correct point?